### PR TITLE
EIP Author's Manual

### PR DESCRIPTION
**Pull request by gurukamath**
First draft of the EIP authors manual. Looking for suggestions for improvement, things that can be added, removed, re-done etc.

### CODE REVIEW

**Comment by poojaranjan** [comment]
Helpful document. I wonder if we should have it as an EIP? cc: @SamWilsn @lightclient

File: EIP_AUTHORS_MANUAL.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+5. **Community Discussion**: The EIP is now ready for discussion with the broader Ethereum community. Although the feedback from the community can be sought at all stages of the EIP lifecycle, we feel that having a reference implementation and tests act as a good bridge between research and client implemtation. It also helps core developers (who have limited time and resources) to understand the EIP better and provide more informed feedback.
+6. **Inclusion in a Fork**: Contigent on the community agreeing to the merits of the EIP, it can be included for implementation in a future fork. The test vectors can then be generated using the reference implementation and made available to the client teams.
+7. **Client Implementation**: The EIP is implemented by the various client teams with the test vectors as the guide.
+8. **Mainnet Deployment**: The EIP is deployed on the Ethereum mainnet.
```
Comment (SamWilsn) [inline EIP_AUTHORS_MANUAL.md:12]: As an EIP Editor, I'm concerned that authors will get these stages confused with the stages of EIPs (draft, review, last call, etc.) We should share terminology whenever possible. How would you feel about something like:

```suggestion
| Stage              | Activities  | Outputs |
| ------------------ | ----------- | ------- |
| _Pre‑Draft_        | Prospective EIP author conceives of an idea for an improvement to Ethereum, and discusses with the community. | <ul><li>Vague Consensus on [Ethereum Magicians][0]</li></ul> |
| **Draft**          | <p>EIP author writes a technical human-language document describing the improvement, initially in broad strokes and becoming more specific over time.</p><p>Concurrently, they develop a Python reference implementation to make the EIP executable and identify any immediate/obvious implementation issues. For example, the EIP may not be compatible with some detail of the current Ethereum Virtual Machine.</p><p>Finally for this stage, the author begins to write test schemes for the EIP. Having the reference implementation should help identify the various logical flows to test and thus feed into more robust testing. Once the test schemes are written, the reference implementation can then be used to fill the tests and generate the test vectors.</p> | <ul><li>Complete (but not final) document in [EIPs Repository][1]</li><li>Reference Implementation in [EELS][2]</li><li>Initial Tests in [EEST][3]</li></ul> |
| **Review**         | <p>The broader Ethereum community discusses and provides input on the proposal.</p><p>Although the feedback from the community can be sought at all lifecycle stages, having a reference implementation and tests act as a good bridge between research and client implemtation. It also helps core developers (who have limited time and resources) to understand the EIP better and provide more informed feedback.</p> | <ul><li>Complete &amp; final document in the [EIPs Repository][1]</li><li>Comprehensive tests in [EEST][3]</li></ul>
| **Last&nbsp;Call** | Usually after being nominated for inclusion in a fork, the EIP author signals that the proposal is effectively done and begins the last period for comments/discussion. | <ul><li>Complete reference implementation in [EELS][2]</li><li>Complete tests in [EEST][3]</li><li>Immutable proposal in [EIPs Repository][1]</li></ul> |
| **Final**          | The proposal is now immutable (cannot be changed) and exists for reference. | <ul><li>Mainnet client implementations</li></ul> |

[0]: https://ethereum-magicians.org/
[1]: https://github.com/ethereum/EIPs/
[2]: https://github.com/ethereum/execution-specs
[3]: https://github.com/ethereum/execution-spec-tests
```

[Rendered preview](https://gist.github.com/SamWilsn/b828d3b393fbf1d75beca9a62892eec7)

File: EIP_AUTHORS_MANUAL.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Folder Structure
+
+# Forks live on mainnet
+
+The folder `src/ethereum` contains the specifications for the different execution layer forks. Each fork has its own folder. For example, the folder `src/ethereum/frontier` contains the specifications for the Frontier hardfork. The `state_transition` function which is available in the `src/ethereum/<FORK NAME>/fork.py` is the transition function for each fork.
+
+# Forks under development
+
+At the time of writing, the Prague Fork is still under development and the previous fork is Cancun, which is live on Mainnet.
```
Comment (SamWilsn) [inline EIP_AUTHORS_MANUAL.md:29]: We will constantly forget to update this. Probably better to omit references to specific forks.

**Comment by SamWilsn** [comment]
@poojaranjan That's a good idea. If EELS reference implementations become mandatory, we'll have to move this to EIP-1.

Let's leave it here for now, with the intent to move it later?

File: EIP_AUTHORS_MANUAL.md (hunk 1 above)
Comment (gurukamath) [reply EIP_AUTHORS_MANUAL.md:12]: Good point. Makes total sense to me.

File: EIP_AUTHORS_MANUAL.md (hunk 2 above)
Comment (gurukamath) [reply EIP_AUTHORS_MANUAL.md]: I'm not sure this needs to be updated at all. I was intending for this to just be a concrete example of the folder structure for the fork under development. Perhaps the fact that this is just an example is unclear from the text. I have updated the text to make that fact more explicit.

**Review by petertdavies — APPROVED** [review]
This looks great! It sets out the proposed process in a really clear and concise manner.

**Pull request merged**